### PR TITLE
Update Kallisto: include missing datatypes and remove unnecesary conditional

### PR DESCRIPTION
**Pull request by gallardoalba**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/kallisto/kallisto_quant.xml [hunk 1]
```diff
@@ -59,33 +59,26 @@
         <conditional name="single_paired">
             <param name="single_paired_selector" type="select" label="Single-end or paired reads">
                 <option value="single" selected="true">Single-end</option>
-                <option value="paired">Paired</option>
+                <option value="paired_single">Paired.end (individual files)</option>
+                <option value="paired_collection">Paired-end (collections)</option>
             </param>
             <when value="single">
-                <param name="reads" type="data" format="fastq,fastq.gz" label="Reads in FASTQ format" />
+                <param name="reads" type="data" format="fastq,fastq.gz,fastqsanger,fastqsanger.gz" label="Reads in FASTQ format" />
```
Comment (bgruening) [inline tools/kallisto/kallisto_quant.xml:66]: fastq should be a superset and include fastqsanger - is that really needed?

File: tools/kallisto/kallisto_quant.xml [hunk 2]
```diff
@@ -59,33 +59,26 @@
...
-                    <option value="--rf-stranded">Strand specific reads, first read reverse</option>
-                </param>
+            <when value="paired_single">
+                <param name="forward" type="data" format="fastq,fastq.gz,fastqsanger,fastqsanger.gz" label="Forward reads" />
+                <param name="reverse" type="data" format="fastq,fastq.gz,fastqsanger,fastqsanger.gz" label="Reverse reads" />
             </when>
+            <when value="paired_collection">
+                <param name="reads" type="data_collection" format="fastq,fastq.gz,fastqsanger,fastqsanger.gz" collection_type="paired" label="Collection of reads" />
+            </when>
+            <param name="lib_type" type="select" label="Library strandness information">
```
Comment (bgruening) [inline tools/kallisto/kallisto_quant.xml:77]: that parameter is not part of any when? I don't think this is possible. You can make a macro out of it and include it in both paried-end cases.

File: tools/kallisto/kallisto_quant.xml (hunk 1 above)
Comment (gallardoalba) [reply tools/kallisto/kallisto_quant.xml:66]: This is perhaps not necessary at all.

File: tools/kallisto/kallisto_quant.xml (hunk 2 above)
Comment (gallardoalba) [reply tools/kallisto/kallisto_quant.xml:77]: Yes you are right.

**Review by bgruening — APPROVED** [review]

**Pull request merged**